config/routes.rb

```diff
@@ -2,6 +2,13 @@
   # The priority is based upon order of creation: first created -> highest priority.
   # See how all your routes lay out with "rake routes".
 
+  root 'shipments#index'
+  get 'estimate/ups/:zip/:weight' => 'ups_shipments#estimate'
+  get 'estimate/usps/:zip/:weight'  => 'usps_shipments#estimate'
+
+  post 'ship' => 'shipments#ship'
```

Comment:
Why did you choose `post` here? Is there too many parameters for the URL length?